feat: add support for bandwidth-sharing techniques

afonsofonbraga opened the pull request.

Commit:
feat: add ptp support

doudou commented on tasks/Task.cpp:
```diff
@@ -453,6 +461,147 @@ string Task::convertFrameModeToPixelFormat(frame_mode_t format, uint8_t data_dep
+            if (device_infos.size() !=
+                static_cast<size_t>(_camera_config.get().ptp_sync.number_of_cameras)) {
+                LOG_WARN_S << "Not all cameras are discovered yet." << endl;
+                system.UpdateDevices(100);
+                device_infos.clear();
+                device_infos = system.GetDevices();
+            }
+            GenICam::gcstring ptp_status;
+            if (device_info.IpAddress() == current_device_ip) {
+                // get PTP status
+                ptp_status =
+                    Arena::GetNodeValue<GenICam::gcstring>(current_device.GetNodeMap(),
+                        "PtpStatus");
+            }
+            else {
+                ArenaDevice dev(system.CreateDevice(device_info), system);
+                ptp_status =
+                    Arena::GetNodeValue<GenICam::gcstring>(dev.device->GetNodeMap(),
+                        "PtpStatus");
+            }
+
+            if (ptp_status == "Master") {
+                if (masterFound) {
+                    // Multiple masters -- ptp negotiation is not complete
+                    restartSyncCheck = true;
+                    break;
+                }
+
+                masterFound = true;
+            }
+            else if (ptp_status != "Slave") {
+                // Uncalibrated state -- ptp negotiation is not complete
+                restartSyncCheck = true;
+                break;
+            }
+        }
+
+        // A single master was found and all remaining cameras are slaves
+        if (!restartSyncCheck && masterFound)
+            break;
```
Create a separate isPTPSyncSettled(device_info) method  that returns true or false and then

```
if (isPTPSyncSettled(device_info)) {
   return;
}
```

Commit:
feat: add support for explicit transfer control

doudou committed:
fix: reset packet and frame tranmission delays in acquisitionConfiguration

doudou committed:
fix: initialize OpenSystem once globally

We otherwise cannot have more than one driver in a single process

doudou committed:
fix: cannot call TransferStart in auto-transfer mode

doudou committed:
fix: stop transfer on exception in explicit transfer mode

Commit:
chore: create TransmissionConfiguration

doudou committed:
fix: cleanup transmission configuration

Commit:
chore: add max number of timeouts and incomplete images

doudou committed:
fix: separate the PTPSync acquisition mode from the delay configuration

This commit refactors PTPSync support to allow to configure PTP
without using PTPSync, and improves PTP configuration.

doudou committed:
fix: do not AcquisitionFrameRateEnable in PTPSync mode

doudou committed:
fix: reset timeout counters on start

doudou committed:
fix: evaluate timeouts/incomplete frames when they happen

The current code is returning from acquireFrame when there is a
timeout or incomplete frame, *before* the timeouts were evaluated.

This caused the timeouts to only be evaluated after a good frame
was received. Move the check in updateHook to fix.

doudou committed:
fix: evaluate timeouts at each updateHook

The current implementation was checking them only after time
crossed m_check_status_deadline. This change makes sure we react
to timeouts sooner, which would be important if we increase the
status period.

Commit:
feat: add info about number of timeouts and incomplete images

Commit:
fix: remove autoset to minimum and maximum values

doudou committed:
fix: do allow setting the packet delay to zero

It is a valid configuration value at least on the Triton, so I don't
see why we should not be allowing it (and have it as default)

doudou reviewed (approved):
Looks good to me.
